src/devices/bus/s100/vectordualmode.cpp

```diff
@@ -101,13 +107,14 @@ uint8_t s100_vector_dualmode_device::s100_sinp_r(offs_t offset)
 		bool seek_complete; // HDD
 		bool loss_of_sync; // HDD
 		if (hdd_selected()) {
+			mfm_harddisk_device *hdd = m_hdd->get_device();
 			write_protect = false;
-			ready = true;
-			track0 = false;
-			seek_complete = true;
+			ready = hdd->ready_r() == ASSERT_LINE;
+			track0 = hdd->trk00_r() == ASSERT_LINE;
+			seek_complete = hdd->seek_complete_r() == ASSERT_LINE;
 			loss_of_sync = true;
 		} else {
-			floppy_image_device* flop = m_floppy[m_drive]->get_device();
+			floppy_image_device *flop = m_floppy[m_drive]->get_device();
```

Comment:
I don’t think this will work properly given the implementation of `hdd_selected`.  If no hard disk is mounted when drive 0 is selected, it will access the floppy drive.  Shouldn’t it fail to access the absent hard disk in that case?

Reply:
I'd say, "I don't know" along with "I don't understand what is the physical analog of that." The physical hard drives didn't have removable platters.

But in this PR I also set floppy0 to `nullptr` by default in `device_add_mconfig`, so either way there's no media. I'll remove the `get_device()->exists()` from `hdd_selected()`.

Reply:
But what happens on real hardware if drive 0 is selected and no hard disk is connected?  I’d imagine it would simply fail to read the hard disk.  However if I’m reading this properly, if that happens here, `hdd_selected()` will return `false` and it will access the floppy drive instead.

Reply:
The board can access either FDD or HDD as disk 0. st506 has mostly the same control signals as shugart/floppy and the card has them act mostly as single bus. There's a configuration area where you cut traces and solder jumper wires to adjust the card to use FDD or HDD. The drive select 0 line is independent between the hdd/fdd connectors.

What I had thought would be ideal was to have a machine configuration choose whether HDD or FDD is selected and to adjust the available slots based on it. But I was still figuring out how that would be done.

I played with this more, and it is seeming the answer is "don't." Always have the slots there, even it would require machine changes for them to be selected. Or make each board configuration a different device class, which I might have done already if Vector Graphic had named the stripped down version.